backend/src/main/java/fk/prof/backend/aggregator/ProfileWorkInfo.java

```diff
@@ -0,0 +1,81 @@
+package fk.prof.backend.aggregator;
+
+import com.koloboke.collect.map.hash.HashObjIntMap;
+import com.koloboke.collect.map.hash.HashObjIntMaps;
+import recording.Recorder;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+/**
+ * Non getter methods in this class are not thread-safe
+ * But, this should not be a concern because instances of this class are maintained for every work id
+ * Updates to state associated with a work id happens in context of a request (single thread)
+ */
+public class ProfileWorkInfo {
```

Comment:
We might, can show more details on the UI. Felt this could be added later (post v1, if we see a need) with a new optional field in aggregated profile's proto.